generate-avatars/Indian135/generate-identicons.py

```diff
@@ -0,0 +1,88 @@
+from PIL import Image,ImageDraw
+import hashlib
+
+def draw_shape(x,y,hexa_num,image,color):
+    # type: (int,int,str,Image,(int,int,int)) -> None
+    num = int(hexa_num,16)
+    draw = ImageDraw.Draw(image)
+    if num == 0:
+        draw.line([x+0,y+0,x+4,y+0],color)
+        draw.line([x+4,y+0,x+4,y+9],color)
+        draw.line([x+4,y+9,x+0,y+9],color)
+        draw.line([x+0,y+9,x+0,y+0],color)
+        draw.line([x+1,y+1,x+3,y+1],color)
+        draw.line([x+3,y+1,x+3,y+8],color)
+        draw.line([x+3,y+8,x+1,y+8],color)
+        draw.line([x+1,y+8,x+1,y+1],color)
+    elif num == 1:
+        draw.line([x+2,y+2,x+2,y+7],color)
+    elif num == 2:
+        draw.polygon([x,y,x+4,y+4,x+4,y+9,x,y+9],color)
+    elif num == 3:
+        draw.polygon([x,y+4,x+4,y,x+4,y+9,x,y+9],color)
+    elif num == 4:
+        draw.polygon([x,y,x+4,y,x+4,y+5,x,y+9],color)
+    elif num == 5:
+        draw.polygon([x,y,x+4,y,x+4,y+9,x,y+5],color)
+    elif num == 6:
+        draw.polygon([x,y,x+4,y,x+4,y+2,x,y+2],color)
+        draw.polygon([x,y+7,x+4,y+7,x+4,y+9,x,y+9],color)
+    elif num == 7:
+        draw.polygon([x,y+3,x+4,y+3,x+4,y+6,x,y+6],color)
+    elif num == 8:
+        draw.line([x+0,y+0,x+0,y+9],color)
+        draw.line([x+1,y+0,x+1,y+9],color)
+        draw.line([x+3,y+0,x+3,y+9],color)
+        draw.line([x+4,y+0,x+4,y+9],color)
+    elif num == 9:
+        draw.line([x+2,y,x+2,y+9],color)
+    elif num == 10:
+        draw.polygon([x,y+2,x+4,y+6,x,y+6],color)
+    elif num == 11:
+        draw.polygon([x+4,y+2,x+4,y+6,x,y+6],color)
+    elif num == 12:
+        draw.polygon([x,y+2,x+4,y+2,x,y+6],color)
+    elif num == 13:
+        draw.polygon([x,y+2,x+4,y+2,x+4,y+6],color)
+    elif num == 14:
+        draw.line([x+0,y+0,x+0,y+9],color)
+        draw.line([x+2,y+0,x+2,y+9],color)
+        draw.line([x+4,y+0,x+4,y+9],color)
+    elif num == 15:
+        draw.polygon([x+0,y+0,x+4,y+0,x+4,y+1,x+0,y+1],color)
+        draw.polygon([x+0,y+4,x+4,y+4,x+4,y+5,x+0,y+5],color)
+        draw.polygon([x+0,y+8,x+4,y+8,x+4,y+9,x+0,y+9],color)
+
+def get_identicon(email_address):
+    # (str) -> None
+    md5 = hashlib.md5()
+    md5.update(email_address.encode('ascii','ignore'))
+    hash_code = md5.hexdigest()
+    
+    bg_color_code = hash_code[:3]
+    fg_color_code = hash_code[3:6]
+    point_code = hash_code[6]
+    # In case background and foreground code are same, the following will avoid problems.
+    if bg_color_code == fg_color_code:
```

Comment:
This part might not be the best way forward. What if they are not the same but still very close so that there is very less contrast? Or, what if the second color chosen also has such a problem?

I would recommend using the inverse of the background color in case of a conflict and recommend measuring conflict as being *near* the bacground color instead of equal. Consider using a colors library or writing your own comparision function for this.

Reply:
If there is a conflict and I inverse the color in that case, than it is possible that some other email's hash may have the same first 3 digits and the next 3 digits may produce the inverse of the color produced by the first 3. If this happens the identicon will be the same (assuming the rest of the digits to be the same).